crypto/dh/dh_key.c

```diff
@@ -316,7 +316,12 @@ static int generate_key(DH *dh)
                 if (dh->length != 0
                     && dh->length >= BN_num_bits(dh->params.p))
                     goto err;
-                l = dh->length ? dh->length : BN_num_bits(dh->params.p) - 1;
+                /*
+                 * If dh->length is not set, use minimum key size supporting
```

Comment:
I note that this appears to be in the "else" clause of an "Is it an approved safe prime?" check (see comment at line 297).

Reply:
Yes, but we are generating safe primes in the dh paramgen by default. Anyway, I am willing to drop this part of the change as arbitrary dh parameter support is something truly legacy.

Reply:
Maybe we should set the length in the paramgen instead?

Reply:
> Maybe we should set the length in the paramgen instead?

Done so.